### ISSUE TITLE
Author unable to activate pre-assigned account

### ISSUE THREAD

**Issue by only1chunts** [gigascience/gigadb-website#1818]
**Describe the bug**
We add new users to GigaDB via the spreadsheet upload, we then ask the authors to activate their account by going to the forgotten password page. On the old version this included a tick box to say they agree to our term and conditions, that tick box appears to be missing from the new version, so they are unable to click it. Consequently the database is not updated with the user is activated "yes" boolean.

The first user I asked to do the account activation wrote back with 

> thanks for creating the account. However, after initializing my password using the password reset 
> option, I can’t login to system. On the login page I got the error message „User is not activated“.

**Comment by rija** [comment]
Hi @luistoptal,
The disappearance of the check box sounds like a collateral from the re-design.
Also, it seems the pages (e.g: /user/welcome/) involved in the new user workflow still harbour the old design.
Would you mind looking at this?

**Comment by luistoptal** [comment]
> new user workflow

I suspect there's a handful of pages I have not seen here. How can I replicate this flow in local dev so I can see the UI?

**Comment by only1chunts** [comment]
Hi Luis,
I'm not sure I understand what Rija is saying about the /user/welcome workflow? so I'll ignore that :-)

The page that has a problem is https://staging.gigadb.org/site/forgot , there needs to be the agree to terms and conditions tick box, the same as appears on the create new user page (https://staging.gigadb.org/user/create)
The action of that tick box should mean the database entry for that user is set to activated=yes

Its probably worthwhile including a similar box for the join mailing list yes/no as well (also as shown on the create user page).

**Comment by luistoptal** [comment]
@only1chunts @rija Regarding the `site/forgot` misisng checkbox, the relevant files seem to be: `protected/views/resetPasswordRequest/forgot.php`, `protected/controllers/ResetPasswordRequestController.php`, `protected/models/ForgotPasswordForm.php`

I see no `terms` attribute in the model so I guess it needs to be added to the backend. Once this is done the checkbox can be added to the frontend. Or that's what it seems to me :) Please let me know if this is accurate @rija

**Comment by only1chunts** [comment]
@luistoptal,
There are probably better ways to model it, but the old version used the GIGADB_USER table "is_activated" boolean to store when a user has both agreed to the T&Cs and confirmed their email address.

@rija,
Ideally we should look at modelling that correctly with a store for both email confirmation and T&Cs, but that will take time, so for expedience is it possible to use the is_activated value as before?

**Comment by luistoptal** [comment]
@only1chunts @rija I can only see that the forgot page uses the ForgotPasswordForm model that has only an email field.

From the FE side I could add a checkbox that is meaningless to the backend but the user needs to check to be able to submit the form. This checkbox can be connected to the backend at a later stage. Only thing to keep in mind is that the ~user accepting the terms would not be stored anywhere in the DB~ terms checkbox input as checked state would not be stored in the DB

Does it sound like an okay solution, even if temporary?

**Comment by only1chunts** [comment]
The issue we need to resolve is making the user status "active" in the database, so if the front end changes are not going to do it, then we need to wait for @Rija for input.

**Comment by rija** [comment]
Hi @only1chunts 

can you let me know the exact link you are sending to the users after their account has been created?

**Comment by only1chunts** [comment]
previously we were sending

> GigaDB is the repository used to host data and tools associated with articles published in GigaScience Press. As part of your <GigaScience / GigaByte> manuscript we will provide an associated GigaDB dataset to host the data and files required for transparency and reproducibility. To this end we have created a GigaDB account on your behalf. Please use the email address you are receiving this email to as your username and request to reset your password here: https://gigadb.org/site/forgot
> 
> Please remember to check your spam mailbox as the confirmation email for password reset tends to go there!

Which does raise another issue, when you are already logged in that page shows an error message :-) but if you are not logged in it does show the reset password form as expected.

**Comment by rija** [comment]
Hi @only1chunts,

TLDR: the link to send users to activate their account is at the end of my comment

If the link being sent to the user is `https://gigadb.org/site/forgot`,
then the view templates referred to by @luistoptal are the corresponding forms template, and they do not mention anything regarding the "terms" checkbox. Looking at the Git history dating back to 2022, the code for password reset never had the functionality to handle terms, 
And it make sense as password reset is meant to be just that. Maybe the code on CNGB for the same endpoint had that, but somehow got lost when we refactored password reset form to fix its security vulnerabilities.

However, there is another controller (UserController) which has an action (actionChangePassword) and a corresponding view (protected/views/user/changePassword.php).  That view does have the Terms checkbox (as well as the Newsletter checkbox).
But to access that form, it requires the user to be logged in as it is the form for changing password, so users have to be logged in, otherwise whose password to change?  (and there is of course the security aspect where only users should change their own password).

In that case , having the "Terms" checkbox in there doesn't make sense. (I understand the Newsletter checkbox as user may want to change their preference, although the change password form is note the place for it, it should be only on the Edit user profile (and that form as "Mailing list subscriber" checkbox)

Furthermore, from looking at `protected/views/user/changePassword.php`, there is no link between the 'terms' variable for the checkbox and the 'is_activated' field in the database, 
so I don't think the "Terms" checkbox is actually doing anything (and yes, there is no `terms` field in the gigadb_user table).

Finally, I've just tested, when I fill the form to create a new user account (/user/create), there is a Terms checkbox that's mandatory to check. After saving, I can see a new row in the gigadb_user table, but the 'is_activated' field is not set (as explained above).

To have the 'is_activated' field set, I have to wait for the email sent to the me to the email I've resistered as a new user and containing a link to click to confirm my account creation: 

https://gigadb.org/user/confirm/key/3291

Only when I click that link, that the 'is_activated' field is set in the database.

(Btw, this is standard workflow for new user onboarding, all websites with accounts works like that)

Although there's a need for a separate ticket for rethinking the "Terms" checkbox (where, when, do we really need it, do we fix the current on that does nothing),

It seems to me that to fix the original problem that motivated this GitHub ticket, you need to send users the link to actually activate their account (not the forgot password one) which is:


https://gigadb.org/user/confirm/key/\<gigadb user id\>


Then when they try to log in, they should click  reset password (I've tried doing that now, and it works fine)

**Comment by rija** [comment]
Hi all,

I'm going to close this ticket as:
* the underlying issue (not being able to activate users created by curators) is no longer problem now the confirmation url is known
* the user signup workflow works fine

However, I've created two tickets to tackle the couple of issues that were identified during the investigation:

* #1891 
* #1892


@luistoptal, there is nothing for you to do on this ticket that I'm closing, but there is work for you in the above ticket (1892) I've just created
